### PR TITLE
Reorder TH statements for ghc 9.0 compatibility

### PR DESCRIPTION
**Pull request by maralorn**
I am uncertain as to why this did build with previous ghc versions. But ghc 9.0 needs this in this order …

### CODE REVIEW

**Comment by emberdart** [comment]
Looks like until this gets merged, this patch is included in nix.

**Comment by maralorn** [comment]
That’s correct. I normally write the patches to fix the build in nixpkgs, but while I am on it I contribute them upstream.

**Comment by emberdart** [comment]
Ah, great. thanks! Wasn't aware it was you. I should probably start doing the same with other tickets I patch before they're accepted.

**Comment by Ericson2314** [comment]
Yeah I have seen this sort of thing before with GHC 9.0. Yes indeed it's weird it ever worked.

**Comment by Ericson2314** [comment]
@maralorn Did you try GHC 9.2?

**Comment by Ericson2314** [comment]
Ugh, no idea why that didn't work. https://github.com/haskell/actions/tree/main/setup says it should.

**Comment by fendor** [comment]
The release notes of GHC 9.0 https://downloads.haskell.org/ghc/9.0.1-alpha1/docs/html/users_guide/9.0.1-notes.html document this change.

Basically:
> Template Haskell splices now act as separation points between constraint solving passes.

That's why order matters now.

**Review by fendor — APPROVED** [review]
LGTM!

**Comment by fendor** [comment]
#43 should unblock this PR

**Comment by maralorn** [comment]
> The release notes of GHC 9.0 https://downloads.haskell.org/ghc/9.0.1-alpha1/docs/html/users_guide/9.0.1-notes.html document this change.
> 
> Basically:
> 
> > Template Haskell splices now act as separation points between constraint solving passes.
> 
> That's why order matters now.

Yeah, I had also read that changelog in the meanwhile. Just forgot to come back and add the explanation here.
The other annoying change affecting some reflex libs is the simplified subsumption change with the required eta expansions.

**Comment by Ericson2314** [comment]
@maralorn can you merge/rebase this so it picked up @fendor's CI fix to fix the commit I added doing additional CI?

**Comment by Ericson2314** [comment]
@maralorn you lost the commit I had added doing 9.0.1. (We need that and @fendor's fix.)

**Comment by Ericson2314** [comment]
I'll just add again.

**Comment by Ericson2314** [comment]
OK excellent!

**Pull request merged**
